src/injected/web/requests.js

```diff
@@ -113,7 +113,7 @@ function start(req, id) {
       log('warn', null, `Unknown responseType "${responseType}", see https://violentmonkey.github.io/api/gm/#gm_xmlhttprequest for more detail.`);
     }
   }
-  encodeBody(details.data)
+  encodeBody(details.data || details.binary)
```

Comment:
This doesn't compute at all. The `data` is an object but `binary` is a boolean. I'll make a proper PR.